pkg/asset/agent/image/agentimage.go

```diff
@@ -131,6 +126,44 @@ func (a *AgentImage) updateIgnitionImg(ignition []byte) error {
 	return nil
 }
 
+// updateIgnitionInfo updates igninfo.json with the current size of
+// the ignition buffer.
+func (a *AgentImage) updateIgnitionInfo(ignitionSize int) error {
+	ignInfoPath := filepath.Join(a.tmpPath, "coreos", "igninfo.json")
+	ignInfoRaw, err := os.ReadFile(ignInfoPath)
+	if err != nil {
+		return err
+	}
+
+	type ignInfo struct {
+		File   string `json:"file"`
+		Length int64  `json:"length"`
+		Offset int64  `json:"offset"`
+	}
+	var info ignInfo
+	err = json.Unmarshal(ignInfoRaw, &info)
+	if err != nil {
+		return err
+	}
+
+	info.Length = int64(ignitionSize)
+	newIgnInfoRaw, err := json.Marshal(info)
+	if err != nil {
+		return err
+	}
+
+	err = os.WriteFile(ignInfoPath, newIgnInfoRaw, 0)
+	if err != nil {
+		return err
+	}
+
+	if ignitionSize > ignitionMaxSize {
+		logrus.Warnf("ignition content length (%d) exceeds default embed area size (%d), increasing final image size", ignitionSize, ignitionMaxSize)
```

Comment:
I'm not sure this is worth more than a debug log. There's no action the user needs to take, so it is certainly not a warning in my view.

Reply:
That was meant just to avoid surprises (to the user - so that he/she is informed that his current configuration is going to increase the final size of the ISO) and help future troubleshooting (for us - so that for any complain about the increased size we could immediately spot the cause).